lib/blockscore/connection.rb

```diff
@@ -0,0 +1,69 @@
+module BlockScore
+  module Connection
+    def get(path, params)
+      request :get, path, params
+    end
+
+    def post(path, params)
+      request :post, path, params
+    end
+
+    def patch(path, params)
+      request :patch, path, params
+    end
+
+    def put(path, params)
+      request :put, path, params
+    end
+
+    def delete(path, params)
+      request :delete, path, params
+    end
+
+    private
+
+    def headers
+      @headers ||= {
+        'Accept' => "application/vnd.blockscore+json;version=#{BlockScore::VERSION.to_i}",
+        'User-Agent' => 'blockscore-ruby/4.1.0 (https://github.com/BlockScore/blockscore-ruby)',
+        'Content-Type' => 'application/json'
+      }
+    end
+
+    def request(method, path, params)
+      unless BlockScore.api_key
+        fail AuthenticationError, {
+          :error => { :message => 'No API key was provided.' }
+        }
+      end
+
+      begin
+        response = execute_request(method, path, params)
+      rescue SocketError, Errno::ECONNREFUSED => e
+        fail APIConnectionError, { :error => { :message => e.message } }
```

Comment:
Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.<br>Style/SignalException: Use `raise` instead of `fail` to rethrow exceptions.